core/spec.md

```diff
@@ -526,11 +572,70 @@ of the existing entity. Then the existing entity can be deleted.
   all other situations `$structure` MUST NOT be included.
 
 - Constraints:
+  - MUST be unique across all entities in the Registry.
   - MUST be a non-empty absolute URL.
   - MUST be a read-only attribute in API view.
 - Examples:
   - `https://example.com/registry/endpoints/ep1`
 
+##### `shortself` Attribute
+
+- Type: URL
+- Description: A unique absolute URL for an entity. This attribute MUST be
+  an alternative URL for the owning entity's `self` attribute. The intention
+  is that `shortself` SHOULD be shorter in length than `self` such that it
+  MAY be used when the length of the URL referencing the owning entity is
+  important. For example, in cases where the size of a message needs to be
+  as small as possible.
+
+  This specification makes no statement as to how this URL is constructed,
+  to which host/path it references, or whether an HTTP request to this URL
+  will directly perform the desired operation or whether it returns an HTTP
+  redirect to the full `self` URL requiring the client to resend the request.
+
+  If an entity is deleted and then a new entity is created that results in
+  the same `self` URL, this specification does not mandate that the same
+  `shorturl` be generated, but it MAY do so.
+
+  This attribute MUST only appear in the serialization if the `shortself`
+  capability is enabled. However, if this capability is enabled, then disabled,
+  and then re-enabled, the `shortself` values MUST retain their original
+  values. In this sense, implementations might create a `shortself` that is
+  static for the lifetime of the entity and the capability controls whether
+  the attribute is serialized or not.
+
+- Constraints:
+  - MUST be unique across all entities in the Registry.
+  - MUST be a non-empty absolute URL referencing the same entity as the `self`
+    URL, either directly or indirectly via an HTTP redirect.
+  - MUST be a read-only attribute in API view.
+  - MUST NOT change during the lifetime of the entity.
```

Comment:
I suspect this could cause problems - migrating from one shortening service to another feels like a reasonable use case... although I can see how anything using this effectively as an ID could be confused by changes.

Worth discussing in the subgroup with folks more knowledgeable than myself?

Reply:
See previous comment.... good catch!